port_ocean/core/event_listener/base.py

```diff
@@ -36,6 +37,29 @@ def _stop(self) -> None:
         """
         pass
 
+    async def _before_resync(self) -> None:
+        """
+        Can be used for event listeners that need to perform some action before resync.
+        """
+        await ocean.app.update_state_before_scheduled_sync()
+
+    async def _after_resync(self) -> None:
+        """
+        Can be used for event listeners that need to perform some action after resync.
+        """
+        await ocean.app.update_state_after_scheduled_sync()
+
+    async def _resync(
+        self,
+        resync_args: dict[Any, Any],
+    ) -> None:
+        """
+        Triggers the "on_resync" event.
+        """
+        await self._before_resync()
+        await self.events["on_resync"](resync_args)
+        await self._after_resync()
```

Comment:
I know its not in much in the scope, but I am a bit afraid of the integration leaving wrong "last" state.
```suggestion
    async def _resync(
        self,
        resync_args: dict[Any, Any],
    ) -> None:
        """
        Triggers the "on_resync" event.
        """
        await self._before_resync()
        try:
            await self.events["on_resync"](resync_args)
        except Exception:
            await self._on_resync_failure()
        else:
            await self._after_resync()
```